src/common/meta/src/wal.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod kafka;
+pub mod options_allocator;
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::error::Result;
+use crate::wal::kafka::{KafkaConfig, Topic as KafkaTopic};
+pub use crate::wal::options_allocator::WalOptionsAllocator;
+
+pub const WAL_OPTIONS_KEY: &str = "wal_options";
+
+/// Wal configurations for bootstrapping meta srv.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Default)]
+#[serde(tag = "provider")]
+pub enum WalConfig {
+    #[default]
+    #[serde(rename = "raft-engine")]
+    RaftEngine,
+    #[serde(rename = "kafka")]
+    Kafka(KafkaConfig),
+}
+
+/// Wal options for a region.
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Default)]
+#[serde(tag = "provider")]
+pub enum WalOptions {
+    #[default]
+    #[serde(rename = "raft-engine")]
+    RaftEngine,
+    #[serde(rename = "kafka")]
+    Kafka { topic: KafkaTopic },
+}
+
+pub type EncodedWalOptions = String;
```

Comment:
It seems there is no need to add type for the serialized JSON.

Reply:
The `WalOptions` is refactored. See: 
https://github.com/GreptimeTeam/greptimedb/blob/de9cdb500794c1a157022087c552d4853356b30a/src/common/meta/src/wal.rs#L49-L56

Such a refactor is guided by: 
https://github.com/GreptimeTeam/greptimedb/blob/de9cdb500794c1a157022087c552d4853356b30a/src/mito2/src/region/options.rs#L68-L72

Reply:
And the `EncodedWalOptions` is removed.